pre_commit/languages/golang.py

```diff
@@ -20,82 +24,139 @@
 from pre_commit.util import rmtree
 
 ENVIRONMENT_DIR = 'golangenv'
-get_default_version = helpers.basic_get_default_version
 health_check = helpers.basic_health_check
 
 
-def get_env_patch(venv: str) -> PatchesT:
+@functools.lru_cache(maxsize=1)
+def get_default_version() -> str:
+    if helpers.exe_exists('go'):
+        return 'system'
+    else:
+        return '1.19.4'  # C.DEFAULT
+
+
+def get_env_patch(
+        venv: str,
+        language_version: str,
+) -> PatchesT:
+    if language_version == 'system':
+        return (
+            ('PATH', (os.path.join(venv, 'bin'), os.pathsep, Var('PATH'))),
+        )
```

Comment:
this is potentially going to install stuff outside of pre-commit's management isn't it?

Reply:
No, since I've set a `GOPATH` later in the code which now I think can be moved to here